yellowbrick/regressor/influence.py

```diff
@@ -0,0 +1,159 @@
+# Visualize the influence and leverage of individual instances on a regression model.
+#
+# Author:   Benjamin Bengfort <benjamin@bengfort.com>
+# Created:  Sun Jun 09 15:21:17 2019 -0400
+#
+# Copyright (C) 2019 The scikit-yb developers
+# For license information, see LICENSE.txt
+#
+# ID: influence.py [] benjamin@bengfort.com $
+
+"""
+Visualize the influence and leverage of individual instances on a regression model.
+"""
+
+##########################################################################
+## Imports
+##########################################################################
+
+import numpy as np
+import scipy as sp
+
+from yellowbrick.base import Visualizer
+from sklearn.linear_model import LinearRegression
+
+
+##########################################################################
+## Cook's Distance
+##########################################################################
+
+class CooksDistance(Visualizer):
+    """
+    Cook's Distance is a measure of how influential an instance is to the computation of
+    a regression, e.g. if the instance is removed would the estimated coeficients of the
+    underlying model be substantially changed? Because of this, Cook's Distance is
+    generally used to detect outliers in standard, OLS regression. In fact, a general
+    rule of thumb is that D(i) > 4/n is a good threshold for determining highly
+    influential points as outliers.
+
+    This implementation of Cook's Distance assumes Ordinary Least Squares regression,
+    and therefore embeds a ``sklearn.linear_model.LinearRegression`` under the hood.
+    Distance is computed via the non-whitened leverage of the projection matrix,
+    computed inside of ``fit()``. The results of this visualizer are therefore similar
+    to, but not as advanced, as a similar computation using statsmodels. Computing the
+    influence for other regression models requires leave one out validation and can be
+    expensive to compute.
+
+    For more see:
+    https://songhuiming.github.io/pages/2016/11/27/linear-regression-in-python-outliers-leverage-detect/
+
+    Note that Cook's Distance is very similar to DFFITS, and that the Cook's Distance
+    scores can be easily transformed to DFFITS scores.
+
+    Parameters
+    ----------
+    ax : matplotlib Axes, default: None
+        The axes to plot the figure on. If None is passed in the current axes
+        will be used (or generated if required).
+
+    influence_threshold : bool, default: True
+        Draw a horizontal line at D(i) == 4/n to easily identify the most influential
+        points on the final regression.
+
+    kwargs : dict
+        Keyword arguments that are passed to the base class and may influence the final
+        visualization (e.g. size or title parameters).
+    """
+
+    def __init__(self, ax=None, influence_threshold=True, **kwargs):
+        super(CooksDistance, self).__init__(ax=ax, **kwargs)
+        self.influence_threshold = influence_threshold
```

Comment:
likely users would want to have a `colors` type arg for the barplot as well as the influence line.

Reply:
This is a stem plot, not a barplot so I added `linefmt` and `markerfmt` - these are the only ways to change the style of the plot including color. To further integrate the stemplot with the visualizer, I've grabbed the style of the baseline and used it style the influence threshold line.